llvm/lib/SYCLLowerIR/SYCLDeviceRequirements.cpp

```diff
@@ -64,6 +65,8 @@ llvm::computeDeviceRequirements(const module_split::ModuleDesc &MD) {
             ExtractUnsignedIntegerFromMDNodeOperand(MDN, I));
       if (!Reqs.ReqdWorkGroupSize.has_value())
         Reqs.ReqdWorkGroupSize = NewReqdWorkGroupSize;
+      if (Reqs.ReqdWorkGroupSize != NewReqdWorkGroupSize)
```

Comment:
On line 61 can we add a check for !MultipleReqdWGSize?   There is no point in checking again if we already know multiple WG sizes are required.

Also can we call Reqs.ReqdWorkGroupSize.reset() after line 69 to keep all the code related to ReqdWorkGroupSize together?